src/relax/analysis/struct_info_analysis.cc

```diff
@@ -994,6 +1003,17 @@ class SymbolicVarCollector : public relax::ExprVisitor,
 
   void VisitStructInfo_(const FuncStructInfoNode* op) final {
     if (op->params.defined()) {
+      // Visit the parameters once to collect bindings, and another
+      // time to collect usages.  Otherwise, a symbolic variable
+      // defined by a later parameter may be treated as undefined when
+      // used by an earlier parameter.
```

Comment:
When does this happen? I think by definition, the symbolic variables are defined at their first appearance in the signature (going left to right)

Reply:
The symbolic variables are defined in their first appearance in the signature, but only if that appearance is sufficient to provide a definition.  If the symbolic variable can be matched to a specific expression in the caller (e.g. ), then it is bound.  If the symbolic variable cannot be matched to a specific expression in the caller, then it is a usage-site instead.  So, `R.Tensor(shape=[16])` binding to `R.Tensor(shape=[sym_var])` implies a binding of `{sym_var: 16}`, but `R.Tensor(shape=[16])` binding to `R.Tensor(shape=[sym_var1 * sym_var2])` cannot provide any definitions.

So, this edge case occurs in cases like the one shown below, where the first occurrence is a usage, and the definition doesn't occur until a later parameter.

```python
def func(A: R.Tensor(shape=[sym_var1*sym_var2]), B: R.Tensor(shape=[sym_var1, sym_var2])):
    #                       ^^^^^^^^                                ^^^^^^^^
    #                       Usage of sym_var1                       Definition of sym_var1
    ...
```

Reply:
Aha, I hadn't thought of that case. Very good.